MekHQ/unittests/mekhq/campaign/parts/equipment/EquipmentPartTest.java

```diff
@@ -1538,7 +1537,7 @@ public void checkWeaponBayWeaponRemovedOthersOkayTest() {
         when(weaponBay.getType()).thenReturn(bayWeaponType);
         List<WeaponMounted> bayWeapons = new ArrayList<>();
         bayWeapons.add(otherMounted);
-        bayWeapons.add(weaponBay);
+        bayWeapons.add(mounted);
```

Comment:
In several cases, this was the problem - the weaponBay was being added as a weapon of itself, instead of `mounted` which is the key equipment being tested.